sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -1023,6 +1023,21 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(MakeDate(Literal(2019), Literal(7), Literal(32)), null)
   }
 
+  test("ANSI mode: creating values of DateType via make_date") {
```

Comment:
We can simplify the tests by merging it with `creating values of DateType via make_date`
```
test("creating values of DateType via make_date") {
  Seq(true, false).foreach { ansi =>
     withSQLConf(SQLConf.ANSI_ENABLED.key -> ansi.toString) {
      ...
    }
   }

  // ansi test
  checkExceptionInExpression...
  ...
  // non-ansi test
  checkEvaluation(..., null)
  ...
}
```